client/src/components/continuous/continuous.js

```diff
@@ -17,18 +17,15 @@ import HistogramBrush from "./histogramBrush";
 import { margin, width, height, createDimensions } from "./util";
 
 @connect(state => {
-  const ranges = _.get(state, "cells.cells.data.ranges", null);
-  const metadata = _.get(state, "cells.cells.data.metadata", null);
-  const initializeRanges = _.get(state, "initialize.data.data.ranges", null);
+  const metadata = _.get(state.controls2.world, "obsAnnotations", null);
+  const ranges = _.get(state.controls2.world, "summary.obs", null);
 
   return {
     ranges,
     metadata,
-    initializeRanges,
-    colorAccessor: state.controls.colorAccessor,
-    colorScale: state.controls.colorScale,
-    graphBrushSelection: state.controls.graphBrushSelection,
-    axesHaveBeenDrawn: state.controls.axesHaveBeenDrawn
+    colorAccessor: state.controls2.colorAccessor,
+    colorScale: state.controls2.colorScale,
+    selectionUpdate: _.get(state.controls2, "obsCrossfilter.updateTime", null)
```

Comment:
Correct, but it is the only way we know when the crossfilter selection is updated.

Because crossfilter is not an immutable POJO, we have to connect it to the components using something that does change.   In this case, the crossfilter has a property called `updateTime` that is guaranteed to be modified *anytime* the crossfilter selection changes.